__init__.py

```diff
@@ -31,22 +42,156 @@
+    if not zinfo_sensor_name_field:
+        click.echo("ZINFO_SENSOR_NAME_FIELD setting is not given!")
+        raise click.Abort
+
     res = requests.get(
         f"{ZINFO_API_BASE_URL}/zi_wsr.svc/JSON/NL.13/?spcid={zinfo_spcid}",
         headers={"Authorization": access_token},
     )
+    now = datetime.now(tz=utc)
     response = res.json()
-    values = response.get('waarden', [])
+    values = response.get("waarden", [])
     current_app.logger.info(f"Got {len(values)} values...")
-    print(f"Example: {values[0]}")
-    # TODO: parse response
+
+    # Parse response
+    df = pd.DataFrame(values)
+    df = df.iloc[::-1]  # switch order of values so that they run from past to present
+    df[zinfo_event_value_field] = pd.to_numeric(df[zinfo_event_value_field])
+    df[zinfo_event_end_field] = (
+        pd.to_datetime(df[zinfo_event_end_field])
+        .dt.tz_localize(ZINFO_TIMEZONE, ambiguous="infer")
+        .dt.tz_convert(utc)
+    )
+    df = (
+        df.set_index([zinfo_event_end_field, zinfo_sensor_name_field])
+        .sort_index()[zinfo_event_value_field]
+        .to_frame()
+    )
+
+    # Convert from meter data per Z-info sensor name (e.g. meterstanden) to time series data per FlexMeasures sensor
+    zinfo_sensor_mapping: dict = current_app.config.get("ZINFO_SENSOR_MAPPING", {})
+    zinfo_sensor_names_received: List[str] = df.index.get_level_values(
+        zinfo_sensor_name_field
+    ).unique()
+    df_sensors = []
+    for zinfo_sensor_name in zinfo_sensor_names_received:
+        df_sensor = df.loc[
+            df.index.get_level_values(zinfo_sensor_name_field) == zinfo_sensor_name
+        ]
+        if zinfo_sensor_name not in zinfo_sensor_mapping:
+            current_app.logger.error(
+                f"Missing Z-info sensor name {zinfo_sensor_name} in your ZINFO_SENSOR_MAPPING config setting."
+            )
+            continue
+        method_kwargs = zinfo_sensor_mapping[zinfo_sensor_name]["pandas_method_kwargs"]
+        for method, kwargs in method_kwargs:
+            df_sensor = getattr(df_sensor, method)(**kwargs)
+        df_sensors.append(df_sensor)
+    df = pd.concat(df_sensors, axis=0)
+
     if not dryrun:
-        # TODO save
-        pass
+        # Save
+        data_source = ensure_data_source(name="Z-info", type="crawling script")
+        sensors = ensure_zinfo_sensors()
+        sensor_dict = {sensor.name: sensor for sensor in sensors}
+        for zinfo_sensor_name in zinfo_sensor_names_received:
+            if zinfo_sensor_name not in zinfo_sensor_mapping:
+                continue
+            sensor_name = zinfo_sensor_mapping[zinfo_sensor_name]["sensor_name"]
+            if sensor_name not in sensor_dict:
+                current_app.logger.error(
+                    f"No sensor set up for Z-info sensor name {zinfo_sensor_name} ..."
+                )
+                continue
+            sensor = sensor_dict[sensor_name]
+            df_sensor = df.query(
```

Comment:
Is this duplicated effort? You already built a list of `df_sensor` objects in the previous loop.